src/cc_toolchain/cc_toolchain_config.bzl

```diff
@@ -191,8 +191,7 @@ def _make_common_features(ctx):
                 actions = [ACTION_NAMES.c_compile, ACTION_NAMES.cpp_compile],
                 flag_groups = [
                     flag_group(
-                        flags = ["-Wall", "-Wextra", "-Wvla", "-Wconversion"] +
-                                ctx.attr.host_compiler_warnings,
+                        flags = ["-Wall", "-Werror"] + ctx.attr.host_compiler_warnings,
```

Comment:
Why not have `-Wvla` and `-Wconversion`?

Reply:
* conversion causes errors in external libraries, so hard to fix
* vla causes errors in simulation/physics, where we have variable length arrays, which is hard to fix

Reply:
actually, I can disable warnings at a more granular level

Reply:
I have disabled errors on specific warnings that cause errors with libraries

Reply:
Just to be clear, you mean variable length arrays that are internal to Box2D or code we own? If it's the later I can try fix it.

Reply:
it's the latter, do you want to fix it as part of this PR?

Reply:
```
In file included from firmware/shared/physics_linear_algebra.c:1:0:
./firmware/shared/physics_linear_algebra.h:13:18: error: ISO C90 forbids variable length array 'matrix' [-Werror=vla]
                  const float matrix[lhs_len][rhs_len]);
                  ^~~~~
./firmware/shared/physics_linear_algebra.h:13:18: error: ISO C90 forbids variable length array 'matrix' [-Werror=vla]
./firmware/shared/physics_linear_algebra.h:25:20: error: ISO C90 forbids variable length array 'matrix' [-Werror=vla]
                    const float matrix[lhs_len][rhs_len]);
                    ^~~~~
./firmware/shared/physics_linear_algebra.h:25:20: error: ISO C90 forbids variable length array 'matrix' [-Werror=vla]
firmware/shared/physics_linear_algebra.c:4:18: error: ISO C90 forbids variable length array 'matrix' [-Werror=vla]
                  const float matrix[lhs_len][rhs_len])
                  ^~~~~
firmware/shared/physics_linear_algebra.c:4:18: error: ISO C90 forbids variable length array 'matrix' [-Werror=vla]
firmware/shared/physics_linear_algebra.c:17:20: error: ISO C90 forbids variable length array 'matrix' [-Werror=vla]
                    const float matrix[rhs_len][lhs_len])
                    ^~~~~
firmware/shared/physics_linear_algebra.c:17:20: error: ISO C90 forbids variable length array 'matrix' [-Werror=vla]
```

Reply:
```
software/simulation/physics/physics_robot_model.cpp:53:40: error: variable length array 'vertices' is used [-Werror=vla]
     b2Vec2 vertices[shape_points.size()];
                                        ^
software/simulation/physics/physics_robot_model.cpp:60:43: warning: conversion to 'int32 {aka int}' from 'std::vector<Point>::size_type {aka long unsigned int}' may alter its value [-Wconversion]
     shape->Set(vertices, shape_points.size());
                          ~~~~~~~~~~~~~~~~~^~
software/simulation/physics/physics_robot_model.cpp: In static member function 'static b2PolygonShape* PhysicsRobotModel::getRobotBodyShapeFrontRight(double)':
software/simulation/physics/physics_robot_model.cpp:81:40: error: variable length array 'vertices' is used [-Werror=vla]
     b2Vec2 vertices[shape_points.size()];
                                        ^
software/simulation/physics/physics_robot_model.cpp:88:43: warning: conversion to 'int32 {aka int}' from 'std::vector<Point>::size_type {aka long unsigned int}' may alter its value [-Wconversion]
     shape->Set(vertices, shape_points.size());
                          ~~~~~~~~~~~~~~~~~^~

```

Reply:
```
software/backend/output/radio/mrf/send_reliable_message_operation.cpp:24:34: error: variable length array 'buffer' is used [-Werror=vla]
         uint8_t buffer[3 + length];
                                  ^
```

Reply:
I'm fine fixing the simulation physics as part of this PR.

However @garethellis0 @EvanMorcom there was talk about possibly later using VLAs in the trajectory planner. Seems like these errors would prevent that. Is that something we'd really want to do later / see benefit in?

VLAs are actually something I thought we might use more for C stuff, but if it's going to fail when built with C++ we should think ahead about if we want to use VLAs at all or not.

Reply:
We can specify different flags for C and C++ IIRC.

Reply:
With regards to the external libraries, I think having something is better then nothing, so if we need to ignore conversion warnings for now it's fine, let's just leave a comment why we are.

Reply:
Fair enough. @jonathanlew can you make a quick ticket to fix our internal uses of VLAS on the C++ side? (which I think is just the simulation physics and mrf stuff since we can handle the firmware stuff with C/C++ specific flags). Should be quick to fix afterwards and then we can re-enable the flag.

Reply:
how?

Reply:
Nah, you just need to have a `c_warnings_feature` and a `cpp_warnings_feature`, and then change the `actions` in each to apply to C/C++. Or even better have a `warnings_feature` that applies to both C/C++ and then a `cpp_warnings_feature` that just adds the vla warnings.